blocks/api/serializer.js

```diff
@@ -45,13 +45,10 @@ export function getSaveElement( blockType, attributes ) {
 		saveElement = createElement( save, { attributes } );
 	} else {
 		saveElement = save( { attributes } );
-
-		// Special-case function render implementation to allow raw HTML return
-		if ( 'string' === typeof saveElement ) {
-			return saveElement;
-		}
 	}
 
+	saveElement = applyFilters( 'blocks.getSaveContent.saveElement', saveElement, blockType, attributes );
```

Comment:
Unrelated to this PR. In the future, we should also make sure that functional components are treated as React components, too:

```js
saveElement = save( { attributes } );
```

Reply:
We are about to introduce new hook `blocks.getSaveContent.saveElement`. Is it meant to exist only for the period of deprecation of `saveElement` which is able to return a string?

Reply:
>Unrelated to this PR. In the future, we should also make sure that functional components are treated as React components, too:

Related: https://github.com/WordPress/gutenberg/pull/3745#discussion_r165008918

It is potentially related to this pull request, as we've simplified all paths which necessitated treating the function component any different than the component class. I think at one point in putting this pull request together I'd tried to convert to `createElement`, but ran into challenges. I'll try to revisit and see if it can be done easily.

Reply:
Yes, I tried the similar in the past but ended with splitting into two methods `getSaveContent` and `getSaveElement` only :(

Reply:
>We are about to introduce new hook `blocks.getSaveContent.saveElement`. Is it meant to exist only until the `saveElement` is being able to return a string? If the answer is yes, then we might want to move all the deprecation logic here instead of using a hook.

I don't have a problem with it sticking around. Related to #3800, it can serve as useful in cases like this one where we want to override the entire behavior of save, not just inject some additional props.

Of course, this opens questions of when and why we add filters. Also feel that we should be proactive about documenting them (expecting that, as with PHP, we'd want [hook documentation pages](https://developer.wordpress.org/reference/hooks/the_content/).